File: frontend/src/api/inquiries.ts
/**
 * 5차 스프린트: 직원 문의 처리 API 클라이언트
 * 8차 스프린트: 답변 CRUD 추가
 * GET   /api/inquiries
 * GET   /api/inquiries/{id}
 * PATCH /api/inquiries/{id}/read
 * POST  /api/inquiries/{id}/answer
 * PATCH /api/inquiries/{id}/answer
 * DELETE /api/inquiries/{id}/answer
 */

import { api } from './client'
import type { InquiryAnswer, StaffInquiryDetail, StaffInquiryListResponse } from '../types'

export function getStaffInquiries(params?: {
  category?: string
  status?: string
  limit?: number
  offset?: number
}): Promise<StaffInquiryListResponse> {
  const q = new URLSearchParams()
  if (params?.category) q.set('category', params.category)
  if (params?.status)   q.set('status',   params.status)
  if (params?.limit)    q.set('limit',    String(params.limit))
  if (params?.offset)   q.set('offset',   String(params.offset))
  const qs = q.toString()
  return api.get(`/api/inquiries${qs ? `?${qs}` : ''}`)
}

export function getStaffInquiryDetail(id: number): Promise<{ inquiry: StaffInquiryDetail }> {
  return api.get(`/api/inquiries/${id}`)
}

export function markInquiryRead(id: number): Promise<{ inquiry: { id: number; status: string; readAt: string | null } }> {
  return api.patch(`/api/inquiries/${id}/read`, {})
}

// ── 8차: 답변 CRUD ────────────────────────────────────────────────────────────

export function createInquiryAnswer(
  inquiryId: number,
  content: string,
): Promise<{ answer: InquiryAnswer; inquiryStatus: string }> {
  return api.post(`/api/inquiries/${inquiryId}/answer`, { content })
}

export function updateInquiryAnswer(
  inquiryId: number,
  content: string,
): Promise<InquiryAnswer> {
  return api.patch(`/api/inquiries/${inquiryId}/answer`, { content })
}

export function deleteInquiryAnswer(inquiryId: number): Promise<void> {
  return api.delete(`/api/inquiries/${inquiryId}/answer`)
}

File: Backend/app/schemas/inquiries.py
"""
5차 스프린트: 직원 문의 처리 Pydantic 스키마
8차 스프린트: 답변(AnswerOut) + 요약 확장 + 아이템 hasAnswer 추가
GET /api/inquiries, GET /api/inquiries/{id}, PATCH /api/inquiries/{id}/read
POST/PATCH/DELETE /api/inquiries/{id}/answer
"""

from datetime import datetime

from pydantic import BaseModel


# ── 답변 작성자 ────────────────────────────────────────────────────────────────

class AnswerAuthorOut(BaseModel):
    id:   int
    name: str
    role: str
    model_config = {"from_attributes": True}


# ── 직원용 답변 ────────────────────────────────────────────────────────────────

class AnswerOut(BaseModel):
    id:        int
    inquiryId: int
    content:   str
    author:    AnswerAuthorOut
    createdAt: datetime
    updatedAt: datetime
    canEdit:   bool = False
    model_config = {"from_attributes": True}


# ── 답변 요청 ──────────────────────────────────────────────────────────────────

class AnswerCreateRequest(BaseModel):
    content: str


class AnswerUpdateRequest(BaseModel):
    content: str


# ── 답변 생성 응답 ─────────────────────────────────────────────────────────────

class AnswerCreateResponse(BaseModel):
    answer:        AnswerOut
    inquiryStatus: str


# ── 기존 스키마 ────────────────────────────────────────────────────────────────

class InquiryGuardianOut(BaseModel):
    id:    int
    name:  str
    phone: str | None = None
    model_config = {"from_attributes": True}


class InquiryResidentOut(BaseModel):
    id:           int
    name:         str
    roomNumber:   str | None = None
    careLevel:    str | None = None
    precautions:  str | None = None
    model_config = {"from_attributes": True}


class StaffInquiryItemOut(BaseModel):
    id:                 int
    category:           str
    preview:            str
    guardianName:       str
    residentName:       str
    residentRoomNumber: str | None = None
    status:             str
    hasAnswer:          bool = False   # 8차 추가
    createdAt:          datetime
    model_config = {"from_attributes": True}


class InquirySummaryOut(BaseModel):
    unread:      int
    needsAnswer: int = 0   # 8차 추가 (READ 카운트)
    answered:    int = 0   # 8차 추가 (ANSWERED 카운트)
    byCategory:  dict[str, int]


class StaffInquiryListResponse(BaseModel):
    total:   int
    summary: InquirySummaryOut
    items:   list[StaffInquiryItemOut]


class StaffInquiryDetailOut(BaseModel):
    id:        int
    category:  str
    title:     str | None = None
    content:   str
    status:    str
    guardian:  InquiryGuardianOut
    resident:  InquiryResidentOut
    answer:    AnswerOut | None = None   # 8차 추가
    createdAt: datetime
    readBy:    int | None = None
    readAt:    datetime | None = None
    model_config = {"from_attributes": True}


class StaffInquiryDetailResponse(BaseModel):
    inquiry: StaffInquiryDetailOut


class InquiryReadOut(BaseModel):
    id:     int
    status: str
    readAt: datetime | None = None
    model_config = {"from_attributes": True}


class InquiryReadResponse(BaseModel):
    inquiry: InquiryReadOut
